porcupine/plugins/directory_tree.py

```diff
@@ -253,28 +277,31 @@ def hide_old_projects(self, junk: object = None) -> None:
         # Settings is a weird place for this, but easier than e.g. using a cache file.
         settings.set_("directory_tree_projects", [str(get_path(id)) for id in self.get_children()])
 
-    def refresh_everything(
+    def refresh(
         self, junk: object = None, *, when_done: Callable[[], None] = (lambda: None)
     ) -> None:
+        """This is called whenever something may have changed and the treeview needs updating.
+
+        All tags are deleted, including tags added by other plugins.
```

Comment:
Should also mention that partial updating with `_update_tags_and_content()` can happen too, when something is opened for example.